mvc/controllers/dependency-injection.rst

```diff
@@ -1,8 +1,87 @@
+
+.. tip:: It's important that you have a good understanding of how ASP.NET 5 implements Dependency Injection (DI). If you haven't already done so, please read `dependency injection  <http://docs.asp.net/en/latest/fundamentals/dependency-injection.html>`_ in `ASP.NET 5 Fundamentals <http://docs.asp.net/en/latest/fundamentals/index.html>`_.
+
+Constructor Injection
+---------------------
+ASP.NET 5's built-in support for constructor-based dependency injection extends to ASP.NET MVC 6 controllers. By simply adding a service type to your controller as a constructor parameter, ASP.NET will attempt to resolve that type using its built in service container. Services are typically, but not always, defined using interfaces. For example, if your application has business logic that depends on the current time, you can inject a service that retrieves the time (rather than hard-coding it), which would allow your tests to pass in implementations that use a set time.
+
+.. literalinclude:: dependency-injection/sample/src/ControllerDI/Interfaces/IDateTime.cs
+  :linenos:
+  :language: c#
+
+Implementing an interface like this one so that it uses the system clock at runtime is trivial:
+
+.. literalinclude:: dependency-injection/sample/src/ControllerDI/Services/SystemDateTime.cs
+  :linenos:
+  :language: c#
+
+With this in place, we can use the service in our controller. In this case, we have added some logic to the ``HomeController`` ``Index`` method to display a greeting to the user based on the time of day.
+
+.. literalinclude:: dependency-injection/sample/src/ControllerDI/Controllers/HomeController.cs
+  :linenos:
+  :language: c#
+  :emphasize-lines: 8,10,12,17-30
+  :lines: 1-31,51-52
+
+If we run the application now, we will most likely encounter an error:
+
+	An unhandled exception occurred while processing the request.
+
+	InvalidOperationException: Unable to resolve service for type 'ControllerDI.Interfaces.IDateTime' while attempting to activate 'ControllerDI.Controllers.HomeController'.
+	Microsoft.Extensions.DependencyInjection.ActivatorUtilities.GetService(IServiceProvider sp, Type type, Type requiredBy, Boolean isDefaultParameterRequired)
+
+This error occurs when we have not configured a service in the ``ConfigureServices`` method in our ``Startup`` class. To specify that requests for ``IDateTime`` should be resolved using an instance of ``SystemDateTime``, add the following line to ``ConfigureServices``:
+
+.. literalinclude:: dependency-injection/sample/src/ControllerDI/Startup.cs
+  :linenos:
+  :language: c#
+  :emphasize-lines: 18
+  :lines: 40-58
+  :dedent: 8
+  
+.. note:: This particular service could be implemented using any of several different lifetime options (``Transient``, ``Scoped``, ``Singleton``, or a single instance). Be sure you understand how each of these scope options will affect the behavior of your service. `Learn more <http://docs.asp.net/en/latest/fundamentals/dependency-injection.html>`_.
```

Comment:
Should also update the `Service Lifetimes and Registration Options` section in http://docs.asp.net/en/latest/fundamentals/dependency-injection.html to remove the mentions of "Instance" because it's not (and never was) a concept of its own.